logger_bench_test.go

```diff
@@ -198,6 +198,26 @@ func BenchmarkAddCallerAndStacktrace(b *testing.B) {
 		}
 	})
 }
+func Benchmark5WithsUsed(b *testing.B) {
+	withBenchedLogger(b, func(log *Logger) {
+		make5ChainedChildLoggers(log).Info("used")
+	})
+}
+
+func Benchmark5WithsNotUsed(b *testing.B) {
+	withBenchedLogger(b, func(log *Logger) {
+		l := make5ChainedChildLoggers(log)
+		runtime.KeepAlive(l)
+	})
```

Comment:
Instead of hard-coding the 5, how about something like this?

```go
func benchmarkWithUsed(b *testing.B, N int, use bool) {
  keys := make([]string, N)
  values := make([]string, N)
  for i := 0; i < N; i++ {
    keys[i] = "k" + strconv.Itoa(i)
    values[i] = "v" + strconv.Itoa(i)
  }

  b.ResetTimers()

  withBenchedLogger(b, func(log *Logger) {
    for i := 0; i < N; i++ {
      log = log.With(String(keys[i], values[i]))
    }
    if use {
      log.Info("used")
    } else {
      runtime.KeepAlive(log)
    }
  })
}
```